tests/func/test_diff.py

```diff
@@ -49,7 +51,10 @@ def test(self):
         out = self.dvc.scm.get_diff_trees(self.a_ref)
         self.assertFalse(out[diff.DIFF_EQUAL])
         self.assertEqual(str(self.a_ref), out[diff.DIFF_A_REF])
-        self.assertEqual(str(self.git.head.commit), out[diff.DIFF_B_REF])
+        self.assertEqual(
+            self.git.git.rev_parse(self.git.head.commit, short=True),
+            out[diff.DIFF_B_REF],
+        )
```

Comment:
My change here made me realize this test is checking that 2 things that are hardcoded to be the exact same thing are the exact same thing. Is it a helpful test? Even before using `rev_parse` the code looked like:
```python
        self.a_ref = self.git.head.commit
        ...
            diff.DIFF_B_REF: str(self.git.head.commit),
        ...
        self.assertEqual(str(self.a_ref), out[diff.DIFF_A_REF])
``` so basically comparing `str(self.git.head.commit)` to itself.

Or maybe I'm not understanding the test? Docstrings in some of these classes would be super informative (:

Reply:
Yeah I think I didn't notice the part where `dvc diff` is called to create `out`. The `diff.DIFF_B_REF` I included is in the class' `self.test_dct` which isn't used in this particular test.

So this test is just equivalent to running `dvc diff HEAD` and just checks that both the `DIFF_A_REF` and `DIFF_B_REF` returned are the same commit hash as HEAD? I don't understand why `self.assertFalse(out[diff.DIFF_EQUAL])` though ): Shouldn't `DIFF_EQUAL` be true?

Reply:
> So this test is just equivalent to running dvc diff HEAD and just checks that both the DIFF_A_REF and DIFF_B_REF returned are the same commit hash as HEAD?

Yes.

> I don't understand why self.assertFalse(out[diff.DIFF_EQUAL]) though ): Shouldn't DIFF_EQUAL be true?

It should be false, since the references are not equal. Unless I'm missing something here myself.